src/components/Dashboard.tsx

```diff
@@ -33,44 +33,82 @@ const Dashboard: React.FC = () => {
 
   usePageTitle(t('PAGE_TITLE_HOME'))
 
+  const dashboardCategories = templatesByCategory.filter(({ templateCategory: { uiLocation } }) =>
+    uiLocation.includes(UiLocation.Dashboard)
+  )
+
   return (
     <div id="dashboard">
       <Header as="h2" content={t('MENU_ITEM_DASHBOARD')} />
-      {templatesByCategory
-        .filter(({ templateCategory: { uiLocation } }) => uiLocation.includes(UiLocation.Dashboard))
-        .map(({ templates, templateCategory: { icon: categoryIcon, title: categoryTitle } }) => (
-          <div key={categoryTitle} className="template-category">
-            <div className="title">
-              {categoryIcon && <Icon size="large" color="grey" name={categoryIcon} />}
-              <Header as="h4">{categoryTitle}</Header>
-            </div>
-            <div className="templates">
-              {templates.map((template) => (
-                <TemplateComponent key={template.code} template={template} />
-              ))}
-            </div>
-          </div>
+      {dashboardCategories.map(({ templates, templateCategory: { icon, title, code } }) => (
+        <CategoryComponent
+          key={code}
+          templates={templates}
+          icon={icon as SemanticICONS}
+          title={title}
+        />
+      ))}
+    </div>
+  )
+}
+
+const CategoryComponent: React.FC<{
+  templates: TemplateInList[]
+  icon: SemanticICONS
+  title: string
+}> = ({ templates, icon, title }) => {
+  const [readyToDisplay, setReadyToDisplay] = useState(
+    templates.some((t) => t.dashboardRestrictions === null)
+  )
+
+  return (
+    <div key={title} className={`template-category${readyToDisplay ? '' : ' hidden-element'}`}>
+      <div className="title">
+        {icon && <Icon size="large" color="grey" name={icon} />}
+        <Header as="h4">{title}</Header>
+      </div>
+      <div className="templates">
+        {templates.map((template) => (
+          <TemplateComponent key={template.code} template={template} setReady={setReadyToDisplay} />
         ))}
+      </div>
     </div>
   )
 }
 
-const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template }) => {
+const TemplateComponent: React.FC<{
+  template: TemplateInList
+  setReady: Dispatch<SetStateAction<boolean>>
+}> = ({ template, setReady }) => {
   const { t } = useLanguageProvider()
   const {
     userState: { currentUser },
   } = useUserState()
-  const [loadedFiltersCount, setLoadedFiltersCount] = useState(0)
+  const [filterCounts, setFilterCounts] = useState<{ [key: string]: number }>({})
```

Comment:
Rather than just keeping *count* of the number of loaded filters, we now need to know *which* ones have loaded, so we can set the template/category to "ready" accordingly.